dash/testing/wait.py

```diff
@@ -0,0 +1,39 @@
+"""Utils methods for pytest-dash such wait_for wrappers"""
+import time
+from dash.exceptions import TestingTimeoutError
+
+
+def until(
+    wait_cond,
+    timeout,
+    poll=0.1,
+    msg="expected condition not met within timeout",
+):  # noqa: C0330
+    end_time = time.time() + timeout
+    while wait_cond():
+        time.sleep(poll)
+        if time.time() > end_time:
```

Comment:
swap the `time.sleep` and `if time > end_time` perhaps? So success right at the end still counts.

Reply:
with a poll freq at `0.1`,  I would argue what a poor performance that edge case could be :),  I will move it down anyway.